eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -418,6 +418,34 @@ object Graph {
 
     }
 
+    case class IgnoredEdges(ignoredLocalEdges: Set[ShortChannelId], ignoredRemoteEdges: Set[(PublicKey, PublicKey)]) {
+      def contains(edge: GraphEdge): Boolean = {
+        edge.balance_opt match {
+          case None => ignoredRemoteEdges.contains((edge.desc.a, edge.desc.b))
+          case Some(_) => ignoredLocalEdges.contains(edge.desc.shortChannelId)
+        }
+      }
+
+      def add(edge: GraphEdge): IgnoredEdges = {
+        edge.balance_opt match {
+          case None => copy(ignoredRemoteEdges = ignoredRemoteEdges + ((edge.desc.a, edge.desc.b)))
+          case Some(_) => copy(ignoredLocalEdges = ignoredLocalEdges + edge.desc.shortChannelId)
+        }
+      }
+
+      def add(edges: IterableOnce[GraphEdge]): IgnoredEdges = {
+        edges.iterator.foldLeft(this)((ignored, e) => ignored.add(e))
+      }
+    }
+
+    object IgnoredEdges {
+      val empty: IgnoredEdges = IgnoredEdges(Set.empty, Set.empty)
+
+      def apply(descs: Set[ChannelDesc]): IgnoredEdges = {
+        IgnoredEdges(descs.map(d => d.shortChannelId), descs.map(d => (d.a, d.b)))
+      }
```

Comment:
This is a bit hackish, since an edge should only be in one of the two sets...why not also include our own `nodeId` in that constructor, so that we know which channels are local (by comparing to `d.a`)?

Our `nodeId` can always be found in our `nodeParams`, so it should be easy to propagate it all the way to here.